actors/builtin/miner/miner_actor.go

```diff
@@ -1642,6 +1619,9 @@ func handleProvingDeadline(rt Runtime) {
 		EventType: CronEventProvingDeadline,
 	})
 
+	// Record whether or not we _have_ early terminations now.
+	hasEarlyTerminations := havePendingEarlyTerminations(rt, &st)
```

Comment:
Accessing the state object from the transaction, outside of the transaction, always concerns me a bit (#509). (it's fine in this case, I just wish we had a better way to deal with this)

Reply:
Ok, after discussion I'm coming round to your way in general, but we agree this one is fine as it is.